ServantPersistent/src/Types.hs

```diff
@@ -0,0 +1,124 @@
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeOperators   #-}
+{-# LANGUAGE DeriveGeneric   #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE DeriveAnyClass  #-}
+{-# LANGUAGE DeriveFunctor  #-}
+{-# LANGUAGE TypeInType  #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE PolyKinds  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE GADTs  #-}
+module Types
+    where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Network.Wai
+import Network.Wai.Handler.Warp
+import Servant
+import Servant.API.Experimental.Auth
+import Servant.Server.Experimental.Auth.Cookie
+import Data.ByteString
+import Data.Proxy
+import GHC.Generics
+import Control.Monad.Except
+import Control.Monad.Reader
+import Data.Serialize
+import Data.Text
+import Database.Persist
+import Database.Persist.Postgresql
+import Data.Time.Clock
+import Control.Lens hiding ((.=))
+import Database.Persist.Sql
+import Database.Persist.TH
+import Data.Type.Equality
+import Data.HashMap.Strict (unions,empty)
+
+data Environment = Test | Devel | Production deriving (Eq, Show)
+
+data CookieError = NotPresent | AuthError AuthCookieException deriving (Eq, Show)
+
+data Config = Config
+    { authSettings :: AuthCookieSettings
+    , randomSource :: RandomSource
+    , serverKey :: ServerKey
+    , environment :: Environment
+    , dbPool :: ConnectionPool
+    }
+
+type App = (ReaderT Config (ExceptT ServantErr IO))
+
+data LoginForm = Login { username :: String
+                       , password :: String
+                       } deriving (Show, Generic, Serialize, FromJSON, ToJSON)
+
+data Session = Session { sessionUser :: String
```

Comment:
not clear what `sessionUser` is used for? Why is it a `String` and not a `User`?